src/validationinterface.h

```diff
@@ -35,6 +36,8 @@ class CValidationInterface {
     virtual void Inventory(const uint256 &hash) {};
     virtual void ResendWalletTransactions(int64_t nBestBlockTime) {};
     virtual void BlockChecked(const CBlock&, const CValidationState&) {};
+    virtual void GetScriptForMining(CScript &script) {};
```

Comment:
Wouldn't it be nice if you could use generate() in a non-wallet environment? Regarding RPC test in a post-wallet-split-off world: this could make sense.

If somone uses the internal miner (which has no productive usage IMO) in a wallet-disable mode he probably uses it for testing only.

I agree with adding a warning to the log. But throwing an error would block testing.